layer/executables/dataset/entrypoint.py

```diff
@@ -34,89 +36,69 @@
 set_has_shown_update_message(True)
 
 
-def _run_assertions(
-    asset_name: str,
-    result: Any,
-    assertions: List[Assertion],
-    tracker: RunProgressTracker,
-) -> None:
-    failed_assertions = []
-    tracker.mark_dataset_running_assertions(asset_name)
-    for assertion in reversed(assertions):
-        try:
-            tracker.mark_dataset_running_assertion(asset_name, assertion)
-            assertion.function(result)
-        except Exception:
-            failed_assertions.append(assertion)
-    if len(failed_assertions) > 0:
-        tracker.mark_dataset_failed_assertions(asset_name, failed_assertions)
-        raise Exception(f"Failed assertions {failed_assertions}\n")
-    else:
-        tracker.mark_dataset_completed_assertions(asset_name)
+def runner(dataset_definition: FunctionDefinition) -> Any:
+    def inner() -> Any:
+        return _run(dataset_definition)
 
+    return inner
 
-def _run(user_function: Any) -> None:
-    settings: LayerSettings = user_function.layer
-    current_project_full_name_ = get_current_project_full_name()
+
+def _run(dataset_definition: FunctionDefinition) -> None:
```

Comment:
i think fabric should be passed down here, so it reads `_run(definition, fabric)`

i understand we get it from environment or default to `local` with
```python
def _get_display_fabric() -> str:
    return os.getenv("LAYER_FABRIC", Fabric.F_LOCAL.value)
```
but it does not mean it should be buried that deep. I prefer to inject env variables as early as possible so the rest of the code can be "normally" tested and does not care about environment.

where is the earliest we could get this and pass it down to the functions which need it (inside `run` i assume)?

Reply:
good point, fixing!

Reply:
actually, this code is changed in this subsequent PR for remote runs: https://github.com/layerai/sdk/pull/322/files#diff-78e468dc28d217a4c73c94b60dee7399ec36940d901420f7d05ecf8d5147d2cf. mind if I fix it there?